core/node/gateway_migrator/src/lib.rs

```diff
@@ -96,7 +107,8 @@ pub async fn current_settlement_layer(
     let inflight_count = storage
         .eth_sender_dal()
         .get_inflight_txs_count_for_gateway_migration(!settlement_mode_from_l1.is_gateway())
-        .await?;
+        .await
+        .unwrap();
```

Comment:
I don't think it's a particularly good idea to add an `unwrap()` here just because we want to inspect returned errors. It'd be better to define an error enum:

```rust
#[derive(Debug, thiserror::Error)]
enum GatewayMigratorError {
    ContractCall(#[from] ContractCallError),
    Internal(#[from] anyhow::Error),
}
```

...and use the latter variant in this case.